app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -10,6 +10,9 @@ def callback
       @user = User.from_omniauth(request.env['omniauth.auth'])
 
       if @user.persisted?
+        # https://github.com/scientist-softserv/palni-palci/issues/633
+        WorkAuthorization.handle_signin_for!(user: @user, scope: params[:scope])
```

Comment:
I think this needs a conditional? will this break on non-params scope calls? remember all other SSO calls come through this same method

Reply:
The logic of `WorkAuthorization.handle_signin_for!` uses the user logging in as the time to add/revoke authorization to works.  If `params[:scope]` is `nil`, then we won't grant authorization to any new works.  And we'll revoke any work authorizations that have expired.

```ruby
  def self.handle_signin_for!(user:, authorize_until: 1.day.from_now, work_pid: nil, scope: nil, revoke_expirations_before: Time.zone.now)
    pids = Array.wrap(work_pid) + extract_pids_from(scope: scope)

    # Maybe we get multiple pids; let's handle that accordingly
    pids.each do |pid|
      begin
	authorize!(user: user, work_pid: pid, expires_at: authorize_until)
      rescue WorkNotFoundError
	Rails.logger.info("Unable to find work_pid of #{pid.inspect}.")
      end
    end

    # We re-authorized the above pids, so it should not be in this query.
    where("user_id = :user_id AND expires_at <= :expires_at", user_id: user.id, expires_at: revoke_expirations_before).pluck(:work_pid).each do |pid|
      revoke!(user: user, work_pid: pid)
    end
  end
```